### PR TITLE
[GSoD] Added new documentation website for gRPC-Gateway

### PR DESCRIPTION
**Pull request by iamrajiv**
Added new documentation website for gRPC-Gateway.

### CODE REVIEW

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:21]: This needs to be updated

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:10]: Lets remove this

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:19]: Should we set this to
```suggestion
baseurl: "/grpc-gateway"
url: "https://grpc-ecosystem.github.io"
```
?

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:27]: Copy this from the other file
```suggestion
exclude: ["run.sh"]
```

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:33]: Remove this

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:87]:
```suggestion
footer_content: 'Copyright &copy; 2020. Distributed by a <a href="https://github.com/grpc-ecosystem/grpc-gateway/tree/master/LICENSE.txt">BSD 3-Clause License.</a>'
```

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:108]: Lets comment this out for now, we should consider comments separately.

File: docs/_includes/comments.html
Comment (johanbrandhorst) [inline docs/_includes/comments.html:19]: Lets remove this from this PR and consider it separately

File: docs/index.md
Comment (johanbrandhorst) [inline docs/index.md:3]: Lets keep the old name
```suggestion
title: gRPC-Gateway
```

File: docs/index.md
Comment (johanbrandhorst) [inline docs/index.md:5]:
```suggestion
description: "Documentation site for the gRPC-Gateway"
```

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Really excited about this, but we can't merge this if it means removing the existing pages. I think I was unclear in my suggestion, do you think you could add back the content :grin:?

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:9]: We should uncomment these right?
```suggestion
baseurl: "/grpc-gateway" # the subpath of your site, e.g. /blog
url: "https://grpc-gateway.github.io" # the base hostname & protocol for your site, e.g. http://example.com
```

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:15]:
```suggestion
exclude: ["run.sh"]
```

File: docs/features.md
Comment (johanbrandhorst) [inline docs/features.md:1]: Not your fault, but lets remove this page, it's in the README anyway.

File: docs/usegotemplates.md
Comment (johanbrandhorst) [inline docs/usegotemplates.md:7]: Does this fix the issue with Go templates?

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Awesome, one more round of changes and we can merge this :heart:.

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml]: Why did you remove this? Shouldn't this be set? https://stackoverflow.com/questions/50775454/what-should-the-baseurl-url-of-the-jekyll-blog

Comment (iamrajiv) [reply docs/_config.yml]: @johanbrandhorst Reason for removing this is that since the GitHub Pages site is currently being built from the /docs folder in the master branch so I thought we do not need `baseurl` but I think we should revert the changes.

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:9]: I think this should be 
```suggestion
url: "https://grpc-ecosystem.github.io/" # the base hostname & protocol for your site, e.g. http://example.com
```

File: docs/.gitignore
Comment (johanbrandhorst) [inline docs/.gitignore:1]: We still need this when running locally, please revert this change

File: docs/.gitignore
Comment (johanbrandhorst) [inline docs/.gitignore:7]: I don't think we should ignore this file, it's important for reproducible builds

**Review by johanbrandhorst — APPROVED** [review]

File: docs/_config.yml
Comment (achew22) [inline docs/_config.yml:69]: Don't include the year in copyright notifications.

**Review by achew22 — APPROVED** [review]

File: docs/_config.yml
Comment (johanbrandhorst) [inline docs/_config.yml:69]:
```suggestion
footer_content: 'Copyright &copy; the gRPC-Gateway Authors. Distributed by a <a href="https://github.com/grpc-ecosystem/grpc-gateway/tree/master/LICENSE.txt">BSD 3-Clause License.</a>'
```

**Review by johanbrandhorst — APPROVED** [review]

**Comment by johanbrandhorst** [comment]
Great work so far @iamrajiv, really excited to see what the users are going to think about this!

**Pull request merged**
